src/engine/Operation.h

```diff
@@ -17,6 +20,80 @@ using std::shared_ptr;
 
 class Operation {
  public:
+  class RuntimeInformation {
+   public:
+    void toJson(std::ostream& out) const {
+      out << "{";
+      out << "\"description\" : \"" << _descriptor << "\",";
+      out << "\"time\" : " << _time << ", ";
+      out << "\"was_chached\" : " << _wasCached << ", ";
+      out << "\"details\" : {";
+      auto it = _details.begin();
+      while (it != _details.end()) {
```

Comment:
This could be a `for()` loop making the `it` iterator local to the block so we don't need the `it2` name. Maybe that too could be a `for` loop

Reply:
I'm using a while loop right now to make the detection of when to print another comma easier. When using a for loop it would have to be of the form `for (auto it = begin(); it != end; ) {... ; ++it ...}` or the iterator would need to be increased by one whenever we check if we are more than one before the end (e.g. `if (it + 1 != end() { ... }`).

Reply:
I wouldn't mind `if (it + 1 != end()) {}` but I understand your reasoning